### PR TITLE
Removes Creatable Video Cameras

### PR DESCRIPTION
**Pull request by Fox-McCloud**
I always said I was going to do this the moment it even became remotely semi-meta for security or command to carry around video-cams.

It was fine for a very long time, but this has started cropping up quite routinely.

Needless to say, video cams are horrendously toxic when used by security; TG had the same experience.

Any officer who carries them is pretty much immune from any shenanigans aside from extreme options--it also makes sec+AI even more omnipresent than they already are. The only real chance to do anything about them is EMP them---which just further entrenches the awful meta of "EMP everything as an antag just in case".

Is it realistic that in 2XXX all spess sec forces would have body/helmet cams? Yes, absolutely. Is it healthy for the game direction or game design? NOPE.

This is one of those things where immersion and realism need to take a back seat to game design and overall balance.

Librarian still has his, of course.

:cl: Fox McCloud
del: Can no longer print video cameras in the autolathe
/:cl:

### CODE REVIEW

**Comment by Darkmight9** [comment]
watch the Hos mug the librarian for his camera now cause muh valids 😏

**Comment by Quantum-M** [comment]
The librarian's video camera should probably be more secure if it's the only one, or have it start on said librarian.

**Comment by SomeSortaSquid** [comment]
I have seen it in action first hand.
As a bored civvie at an entertainment monitor, I have seen vampires unable to lie a finger on any officer with a video camera. Response time and precision were deadly to the point where officers could safely patrol maintenance alone, something that shouldn't be possible against vampires.

I don't like the solution, because it's a fun niche item that will fall victim to powergaming. It is a valid solution nonetheless and it will stop a powergamey meta from gaining more popularity.

**Comment by MRgaty** [comment]
Yeah I think is time to stop this meta

**Comment by Carthusia** [comment]
Is it really metagaming though for sec officers to wear body cameras? That's kind of what it amounts to

**Comment by Fox-McCloud** [comment]
>Is it really metagaming though for sec officers to wear body cameras? That's kind of what it amounts to

Is it metagaming that people could, once upon a time, clone death alarms, then implant all of security with them to protect them? (no, this wasn't an exploit)

No. Missing the point; it's an egregiously toxic and poor design for the overall health of the game that's just jamming a thumb in antags' eyes.

**Comment by Carthusia** [comment]
Thus far I haven't really seen any "toxic" activity because of the camera, it would be like removing the RCD from the autolathe and limiting it to the CE,  because people can quickly RCD down walls to get access to locations bypassing the need for ID cards, Emags and what not.

It's just another tool in peoples box and is up to them how they use it, counter it or otherwise bypass it.

**Comment by Spacemanspark** [comment]
While I haven't seen this becoming a major issue or really prevalant recently, I'm fine with this as long as the librarian still has theirs (as stated in the OP).

**Comment by Carthusia** [comment]
> While I haven't seen this becoming a major issue or really prevalant recently, I'm fine with this as long as the librarian still has theirs (as stated in the OP).

I just feel if this isn't a major issue, why would it then need to be removed?

**Comment by Spacemanspark** [comment]
Why does it need to stay

Why do you _need_ a camera unless you're the librarian?
Security certainly doesn't need to have cameras focused on them 24/7. If you're so worried about not being revived after getting ganked in maintenance, then either travel with a buddy, get good, or don't play security. Part of the interest in the game is sometimes not knowing what's going on at all times. 

And, mind you, I wasn't very active on the server during February for life reasons, so when I say 'I haven't seen it becoming an issue', I quite literally mean I haven't seen anything going on. I never said it wasn't an issue going on at all.

**Comment by Rurik123** [comment]
Don't really see hows this is anymore "horrendously toxic" than borgs placing AI xray cams in maint on green or RnD/Genetics distributing xrays and such on green, both of which—correct me if I'm wrong—are relatively accepted here. Only seen it a hand-full of times, and one of which, might I mention, me and two other officers were killed aboard the mining station with our cams on.

It took roughly 20 mins till a new arrival HoS checked the entertainment monitor to see half his force dead.

While I'm aware the incompetence of others isn't an argument for keeping this, I'd argue this isn't as big of a deal as its being made out to be compared to other gamey/toxic things that are done commonly. Compared to those, this feels more like a neat project for the occasional creative warden to undertake.

Personally the last thing I'm worried about when going after a officer, as antag, is a camera. Go into bag, toss the cam away, and if you did it right to prevent a call for help, it's unlikely someone was watching that exact officers cam at that exact time.

**Comment by Carthusia** [comment]
> Why does it need to stay
> 
> Why do you _need_ a camera unless you're the librarian?
> Security certainly doesn't need to have cameras focused on them 24/7. If you're worried about not being revived after getting ganked in maintenance, then either travel with a buddy, don't charge into maintenance alone, or don't play security. Part of the interest in the game is sometimes not knowing what's going on at all times.

To be fair the argument can be made of why the librarian would need it either as their job primarily focuses on books and photos, but in the end its the same answer for either situation: 
Because they can, and because its another tool for a player to choose to use or not depending on their playstyle. 
I know of security officers who REFUSE to turn their suit sensors on because that's how they choose to play with the tools at their disposal, they have the trackers as their tool, and they have chosen not to use them.

> 
> And, mind you, I wasn't very active on the server during February for life reasons, so when I say 'I haven't seen it becoming an issue', I quite literally mean I haven't seen anything going on. I never said it wasn't an issue going on at all.

I feel its not as big a problem as its being out to be and that it may be an overreaction to the perceived issue at hand. 

Instead of potentially mitigating changes such as Protolathe locking it behind a tech level and monitoring the situation for an extended period of time while keeping records, the leap was made for complete removal of a tool, which seems excessive.

Removing tools from players toolbox should never be done in haste, which i in all honesty feel this is being done in due to how no mitigating factors have been put forward thus far that I am aware of.

**Comment by lolzyking** [comment]
I will miss them, it was fun building myself a stack of cameras at the syndie depot, meeting them out with a mass driver and a madcap while disguised as " bomb cam" then waiting for miners/ashwalkers to get close to the ear crime and detonate it.

**Comment by Quantum-M** [comment]
> To be fair the argument can be made of why the librarian would need it either as their job primarily focuses on books and photos...

I am pretty sure the reason librarian gets the camera is due to their alt title of reporter.

I barely ever have seen myself the camera used for legitimate reasons (e.g. reporting stuff to the crew as some sort of journalist) the most I ever seen it being used was with that gateway cheese to see what is in the gateway without the risk of getting stuck in it.

**Comment by farie82** [comment]
I agree with this change but it kinda is the same as engineers placing cameras in maint. Or sec having xray. You lose the element of surprise in all three cases

**Comment by SomeSortaSquid** [comment]
Placing cameras in maints is extremely tedious, but stupidly powerful. It's done mostly by rather bored engineers and takes a while. I don't like it, because it most serves validhunting AIs.
X-rays are a broken anti-fun mechanic. Normally should be used only against vampires, because of their innate x-ray. For both security and vampires, it comes into play later into the round.

Video cameras are a cheap roundstart buff for officers. Suddenly, even a simple ":s help" gives their exact location, possibly not even needed if there's an appointed "overwatch" officer.
The meta seems to still be forming, so it's not very common yet, but even I used to encourage officers to carry those. Then I have realized officers are completely untouchable by antags, which _is_ horrible gameplay.

**Comment by farie82** [comment]
Alternatively. Make it only work when it is in your hands or not on a mob. Can't put it in a box which is located in your bag then

**Comment by Rurik123** [comment]
> Placing cameras in maints is extremely tedious, but stupidly powerful. It's done mostly by rather bored engineers and takes a while. I don't like it, because it most serves validhunting AIs.
> X-rays are a broken anti-fun mechanic. Normally should be used only against vampires, because of their innate x-ray. For both security and vampires, it comes into play later into the round.
> 
> Video cameras are a cheap roundstart buff for officers. Suddenly, even a simple ":s help" gives their exact location, possibly not even needed if there's an appointed "overwatch" officer.
> The meta seems to still be forming, so it's not very common yet, but even I used to encourage officers to carry those. Then I have realized officers are completely untouchable by antags, which _is_ horrible gameplay.

I understand what you mean, and agree on most aspects. Definitely wouldn't want this happening every round, but I doubt it will get to that point, personally. I find it hard, however, to see this making officers untouchable in any way. Out of the three things you mentioned, cams in maint, xrays, and cams on officers, Id argue cams on officers it the lesser of the three evils in the first place.

Out of the things that can be done, such as 5+ secborgs with vtec spam, gygax/phazon spam, cams in maint, xrays, CNS implants, anti-drop, and so on, I'd be rather relaxed if it's just cams. 

Should note I'm not really against this PR, I just wanted to point out body cams aren't _that_ cancerous considering all the other things sec can do to make antags life hell.

**Comment by datlo** [comment]
While I agree with removal, as farie said perhaps have the camera only works when wielded instead. Make it two handed so there's a serious downside to carrying a camera around?

**Comment by Carthusia** [comment]
Another alternative to removal is protolathe locking it, that way its still an option but requires departments to work for it

**Comment by MRgaty** [comment]
> Another alternative to removal is protolathe locking it, that way its still an option but requires departments to work for it

Why do you want a camera so badly you want to put it in a lockbox like a weapon instead of just removing it.

**Comment by Carthusia** [comment]
> Why do you want a camera so badly you want to put it in a lockbox like a weapon instead of just removing it?
> I have never seen it being used for anything more than powergaming.

I never said to put it in a lockbox?

I am against the idea of removing it because overall I dislike the idea of removing a tool from the players toolbox without due cause.

Simply because some players use the tool in question in less orthodox ways does not mean it requires removal.

Again i note: It would be like removing buildable RCD's and limiting them only to the CE's starter RCD because people can use them to bypass doors, walls and be used for other means besides just construction, negating the use of say an emag, cham kit or other syndicate tools used for the same purpose.

It's all about choice, if you slowly remove the tools players can use, overtime things become pretty limited, which is something I feel we can all agree should be avoided were possible.

**Comment by Spacemanspark** [comment]
It's a camera meant for entertainment purposes, not for security to monitor each other.
If we're at the point where we're locking something meant as _entertainment_ rather than a tool behind protolathe research restrictions I think there's an underlying issue.

**Comment by Carthusia** [comment]
> It's a camera meant for entertainment purposes, not for security to monitor each other.
> If we're at the point where we're locking something meant as _entertainment_ rather than a tool behind protolathe research restrictions I think there's an underlying issue.

Personally i don't think we are at that point, it's just the argument was made to remove it, and if people were steadfast on doing so, i merely offered an alternative


> Repurposing things in the game for other useful things is something to be encouraged, mind, makes things interesting.
> But when it begins to ruin the target atmosphere of the game (again, to the point we lock it away behind research [which doesn't really even make in character sense if you think about it, why the fuck is a simple video camera a hassle to research in that case, fuck you could probably buy a dozen a dime at space Walmart]) steps should be taken to reduce the impact.

Steps taken to mitigate a perceived issue due to an item are one thing, a giant leap to removing said item is another. My point is still: If it is perceived to be such an issue, why not protolock it for a bit, test and monitor, judge how it is used and then determine if it still is, or ever was an issue, before deciding to remove a tool from the players toolbox

> I see little issue with this PR. The librarian still has theirs for their alt title job of journalist, and it should stay that way.

Side Note: Is the camera viewable from those entertainment consoles in the bar, barbershop etc, and if so, can people viewing said camera hear what's being said from those in the footage? 

If its not, and this does go through, they really should be able to watch them on the TV and hear what's being said, otherwise it's not really much of a use to the journalist/reporter in their reporting if they wanted to broadcast a live interview with someone.

**Comment by Spacemanspark** [comment]
My point about nonsensically locking an entertainment camera behind a research level stands.

It isn't meant to be a tool in security's toolbox

**Comment by Carthusia** [comment]
> My point about nonsensically locking an entertainment camera behind a research level stands.

The protolathe already contains a few items that don't make to much sense being protolathe locked:
Empty Emergency Oxygen Tank
Empty Oxygen Tank
Light Replacer
Safety Muzzle "Not to be confused with a shock muzzle"

My reasoning for putting forward the protolathe tech level is as an alternative to simply removing the item itself from production.



> It isn't meant to be a tool in security's toolbox

Why wouldn't it be?
 
Soap/clownfish/nanners are not apart of their gear loadout, but they are still free to find and carry some should the player feel like it. 

The same can be said of various other tools, just because they are not assigned to their jobs loadout at the start, does not mean a player cannot make said items apart of their loadout as the round progresses.
 Should the camera be protolathed, this puts such an advantage it provides on par with other items that provide advantages, requiring time, resources and departmental cooperation.

**Comment by MRgaty** [comment]
I like sparks idea of not being able to film inside a backpack

**Comment by Carthusia** [comment]
> I like sparks idea of not being able to film inside a backpack

I like that to, cant broadcast what it cant see, could apply to pockets as well, though not if its pinned to the belt/suit storage slot of a piece of clothing if it is able to do so, this also then would add a bit of a "This or that" scenario, do i use my belt slot for a body camera, or do i keep my belt but not get a camera

File: code/game/jobs/job/support.dm [hunk 1]
```diff
@@ -420,6 +420,8 @@
 	r_pocket = /obj/item/barcodescanner
 	l_hand = /obj/item/storage/bag/books
 	pda = /obj/item/pda/librarian
+	backpack_contents = list(
+		/obj/item/videocam = 1)
```
Comment (AffectedArc07) [inline code/game/jobs/job/support.dm]: I would put this bracket on a newline so it fits in with other instances of `backpack_contents`

File: code/modules/research/designs/autolathe_designs.dm [hunk 2]
```diff
@@ -872,14 +872,6 @@
...
 	category = list("initial", "Miscellaneous")
 
-/datum/design/videocam
-	name = "Video Camera"
-	id = "videocam"
-	build_type = AUTOLATHE
-	materials = list(MAT_METAL = 120, MAT_GLASS = 60)
-	build_path = /obj/item/videocam
-	category = list("initial", "Miscellaneous")
-
```
Comment (AffectedArc07) [inline code/modules/research/designs/autolathe_designs.dm]: Possibly comment this out incase it needs to be easily added back in the future for whatever reason, but at the same time its only 7 lines.

**Pull request merged**